app/main.py

```diff
@@ -1,3 +1,56 @@
-def shop_trip():
-    # write your code here
-    pass
+from collections import defaultdict
+
+from app.customer import Customer
+from app.load_jsons import load_complete_json
+from app.shop import Shop, shop_visit_price_calculation, checkout_for_products
+
+
+def shop_trip() -> None:
+    data = load_complete_json("app/config.json")
+    customers = [Customer(**customer_) for customer_ in data.get("customers")]
+    shops = [Shop(**shop) for shop in data.get("shops")]
+    fuel_price = data.get("FUEL_PRICE")
+
+    for customer in customers:
+        best_price_and_shop = defaultdict()
+        print(f"{customer.name} has {customer.money} dollars")
+        for shop in shops:
+            total_price_and_shop = shop_visit_price_calculation(customer,
+                                                                shop,
+                                                                fuel_price)
+            print(f"{customer.name}'s trip to the {shop.name} "
+                  f"costs {total_price_and_shop['total_price']}")
+
+            if (not best_price_and_shop or total_price_and_shop["total_price"]
+                    < best_price_and_shop["total_price"]):
+                best_price_and_shop = total_price_and_shop
+
+        best_shop_visit_price: float = best_price_and_shop["total_price"]
+        best_shop_instance: Shop = best_price_and_shop["shop"]
+
+        if customer.money >= best_shop_visit_price:
+            print(f"{customer.name} rides to {best_shop_instance.name}\n")
+        else:
+            print(f"{customer.name} doesn't have enough "
+                  f"money to make a purchase in any shop")
+            continue
+
+        print("Date: 04/01/2021 12:33:41")
```

Comment:
![NVIDIA_Share_Roicd1s2Ug](https://github.com/mate-academy/py-shop-trip/assets/13889031/55f37722-862f-4d2e-8fc6-8350da8688f9)
I mean, my way might not be the best but when I tried to use ```datetime.datetime.strftime(datetime.datetime.now(), "%d/%m/%Y %H:%M:%S")``` test were giving me some error that I didn't understand how to handle because I receive same string with this code. But on the other hand - ```datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")``` this line of code gives absolutely same result, and works..

Reply:
> well datetime.now() should work remove hardcoded string please

already did, there are lots of work to fix so I'll probably commit everything only on weekend